_gsocproposals/2020/proposal_CMSSW-ROOT-Mesh-Allocator.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Enable the Mesh Memory Allocator in ROOT
+layout: gsoc_proposal
+project:
+  - CMSSW
+  - ROOT
+year: 2020
+organization: princeton
+---
+
+## Description
+
+Mesh is a drop in replacement for malloc that compacts the heap without rewriting application pointers. More information can be found [here](https://github.com/plasma-umass/Mesh#readme)
+
+According to the authors of the [paper](https://github.com/plasma-umass/Mesh/raw/master/mesh-pldi19-powers.pdf) using the Mesh allocator reduced the memory consumption of Firefox by 16% and Redis by 39%.
+
+
+## Task ideas and expected results
+
+The main task is to understand how the Mesh allocator performs in ROOT. We should start with enabling the allocator for ROOT on UNIX using the LD_PRELOAD mechanism. Then we should produce a set of performance measurements and compare them to the baseline ROOT. This can be done in the context of the rootbench infrastructure. We should understand the bottlenecks if any, and report and fix them either to the ROOT or the Mesh project.
```

Comment:
add a link for rootbench

Reply:
"We should understand the bottlenecks if any, and report and fix them either to the ROOT or the Mesh project." -> We should understand the bottlenecks if any and fix them, reporting to either ROOT or Mesh projects.